test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -2850,17 +2873,21 @@ def forward(self, x):
     def test_random_like_dtype(self):
         class RandNLike(torch.nn.Module):
             def forward(self, x):
-                return torch.mul(
-                    x.to(torch.double), torch.randn_like(x, dtype=torch.double).size(0)
+                return (
+                    x.to(torch.float32)
+                    * torch.randn_like(x, dtype=torch.double)
+                    * torch.tensor(0, dtype=torch.float32)
```

Comment:
The previous test is ineffective because when a value (here is a Python int scalar) is multiplied by a double i.e. `x`, it will be implicitly casted to double. Therefore, the resulting dtype will be irrelevant to what we want to test.

Reply:
Could you help me understand the expected behavior better: is the expected output supposed to be double, but is currently being exported as float32 instead, with this setup, because the last node is float32?

I would also update the test name to reflect the expected behavior. Thanks!

Reply:
Yes, your understanding is correct. I've just added comments indicating the expected dtype.